File: numberOfEquivalentDominoPairs.cs
public class Solution {
    public int NumEquivDominoPairs(int[][] dominoes) {
        var map = new Dictionary<int, int>();
        int count = 0;

        foreach (var d in dominoes) {
            int a = Math.Min(d[0], d[1]);
            int b = Math.Max(d[0], d[1]);
            int key = a * 10 + b;

            if (map.ContainsKey(key)) {
                count += map[key];
                map[key]++;
            } else {
                map[key] = 1;
            }
        }

        return count;
    }
}

File: numberOfEquivalentDominoPairs.py
from typing import List
from collections import defaultdict

class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        count = 0
        freq = defaultdict(int)

        for a, b in dominoes:
            key = tuple(sorted((a, b)))
            count += freq[key]
            freq[key] += 1

        return count
